aquacomputer_d5next.c

```diff
@@ -1710,6 +1734,13 @@ static int aqc_raw_event(struct hid_device *hdev, struct hid_report *report, u8
 	/* Special-case sensor readings */
 	switch (priv->kind) {
 	case aquaero:
+		/* Read hardware version (for v5: 5600, for v6: 6000, usually) */
+		priv->aquaero_hw_version = get_unaligned_be16(data + AQUAERO_HARDWARE_VERSION);
+		if (priv->aquaero_hw_version / 1000 == 5)	/* Get the first digit */
+			priv->aquaero_hw_kind = aquaero5;
+		else if (priv->aquaero_hw_version / 1000 == 6)
+			priv->aquaero_hw_kind = aquaero6;
```

Comment:
There seems to be a small misunderstanding here. I was actually thinking about something like this:
```
switch (priv->aquaero_hw_version) {
case AQUAERO_5_HW_VER:
    priv->aquaero_hw_kind = aquaero5;
    break;
case AQUAERO_6_HW_VER:
    priv->aquaero_hw_kind = aquaero6;
    break;
default:
    priv->aquaero_hw_kind = unknown;
}
```
with defines for the currently known hardware versions.

Reply:
Hmm, alright. I'll switch to that, as I can't find any info on subrevisions of the Aquaero 5 anywhere (which was the idea behind checking only the first number).